--- configure.ac.orig	2010-03-24 21:27:49.000000000 +0100
+++ configure.ac	2010-04-22 01:41:23.000000000 +0200
@@ -747,6 +747,13 @@
 
 ### GraphicsMagick++
 
+WANT_MAGICK=false
+AC_ARG_ENABLE(graphicsmagick,
+  [AS_HELP_STRING([--enable-graphicsmagick],
+     [reading images (default is no)])],
+  [if test "$enableval" = yes; then WANT_MAGICK=true; fi], [])
+
+if $WANT_MAGICK; then
 warn_magick="GraphicsMagick++ library not found.  The imread function for reading image files will not be fully functional."
 
 MAGICK_CPPFLAGS=
@@ -781,6 +788,9 @@
   CPPFLAGS="$save_CPPFLAGS"
   LIBS="$save_LIBS"
 ])
+else
+  warn_magick="GraphicsMagick support disabled."
+fi
 
 if test -z "$warn_magick"; then
   AC_DEFINE(HAVE_MAGICK, 1, [Define if GraphicsMagick++ is available.])
@@ -796,9 +806,15 @@
 # ---------------------------------------------------------------------
 
 ## libraries needed for graphics
+WANT_NATIVE_GRAPHICS=false
+AC_ARG_ENABLE(native-graphics,
+  [AS_HELP_STRING([--enable-native-graphics],
+     [experimental support for native graphics (default is no)])],
+  [if test "$enableval" = yes; then WANT_NATIVE_GRAPHICS=true; fi], [])
 
-warn_freetype=""
+warn_freetype=
 
+if $WANT_NATIVE_GRAPHICS; then
 OCTAVE_OPENGL
 if test -n "$OPENGL_LIBS"; then
   AC_DEFINE(HAVE_OPENGL, 1, [Define if OpenGL is available])
@@ -809,6 +825,9 @@
     XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FT2_CFLAGS"],
     [warn_freetype="FreeType library not found.  Native renderer will not have on-screen text"])
 fi
+else
+  warn_freetype="No FreeType support. Native graphics disabled."
+fi
 
 ## fontconfig library
 
@@ -852,11 +871,13 @@
 
 AC_PATH_PROG(FLTK_CONFIG, fltk-config, no)
 
-warn_fltk_config=""
-warn_fltk_opengl=""
+warn_fltk_config=
+warn_fltk_opengl=
 
 if test "$FLTK_CONFIG" = "no" ; then
   warn_fltk_config="FLTK config script not found.  Native graphics will be disabled."
+elif ! $WANT_NATIVE_GRAPHICS; then
+  warn_fltk_config="No FLTK support. Native graphics disabled."
 else
   FLTK_CFLAGS="`$FLTK_CONFIG $fltkconf_args --use-gl --cflags`"
   FLTK_LDFLAGS="`$FLTK_CONFIG $fltkconf_args --use-gl --ldflags`"
